docs/usage/dns-search-path-optimization.md

```diff
@@ -0,0 +1,113 @@
+
+## DNS Search Path/`ndots` Problem in Kubernetes
+
+As the DNS search path is long and `ndots` is large, a lot of DNS queries might traverse the DNS search path. This results
+in an explosion of DNS requests.
+
+For example, consider the name resolution of the default kubernetes service `kubernetes.default.svc.cluster.local`. As this
+name has only four dots it is not considered a fully qualified name according to the default `ndots=5` setting. Therefore,
+the DNS search path is applied resulting in the following queries being created
+ - `kubernetes.default.svc.cluster.local.some-namespace.svc.cluster.local`
+ - `kubernetes.default.svc.cluster.local.svc.cluster.local`
+ - `kubernetes.default.svc.cluster.local.cluster.local`
+ - `kubernetes.default.svc.cluster.local.network-domain`
+ - ...
+
+In IPv4/IPv6 dual stack systems, the amount of DNS requests may even double as each name is resolved for IPv4 and IPv6.
+
+## General Workarounds/Mitigations
+
+Kubernetes provides the capability to set the DNS options for each pod (see
+[Pod DNS config](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config) for details).
+However, this has to be applied for every pod (doing name resolution) to resolve the problem. A mutating webhook may be
+useful in this regard. Unfortunately, the DNS requirements may be different depending on the workload. Therefore, a general
+solution may difficult to impossible.
+
+Another approach is to use always fully qualified names and append a dot (`.`) to the name to prevent the name resolution
+system from using the DNS search path. This might be somewhat counterintuitive as most developers are not used to the
+trailing dot (`.`). Furthermore, it makes moving to different landscapes more difficult/error-prone.
+
+## Gardener specific Workarounds/Mitigations
+
+Gardener allows users to [customize their DNS configuration](custom-dns.md). CoreDNS allows several approaches to deal with
+the requests generated by the DNS search path. [Caching](https://coredns.io/plugins/cache/) is possible as well as
+[query rewriting](https://coredns.io/plugins/rewrite/). There are also several other [plugins](https://coredns.io/plugins/)
+available, which may mitigate the situation.
+
+## Gardener DNS Query Rewriting
+
+As explained [above](#dns-search-path-ndots-and-kubernetes), the application of the DNS search path may lead to the undesired
+creation of DNS requests. Especially with the default setting of `ndots=5`, seemingly fully qualified names pointing to
+services in the cluster may trigger the DNS search path application.
+
+Gardener allows to automatically rewrite some obviously incorrect DNS names, which stem from application of the DNS search
+path, to the most likely desired name. The feature can be enabled by setting the Gardenlet feature-gate `CoreDNSQueryRewriting` to `true`:
```

Comment:
I know that we currently don't have "target audience/persona" (end user (shoot owner), gardener operator) for each guide but I think it still makes sense to pay attention to it. If we assume that this guide is for end users (shoot owners) the this section is not applicable for them - they cannot enable the feature gate.
We could move this feature gate part to a separate guide for gardener operators - in this guide we could also shortly mention the `ShootDNSRewriting` admission plugin - its purpose, how to enable, how to configure, etc. WDYT?

Reply:
In general, I agree that it is better to have a clear target audience. However, I am unsure whether this would be a good topic to start.
From my perspective, the biggest challenge regarding the DNS query rewriting is to raise awareness/understanding of the problem. I think that you need to understand the problem regardless whether you are an operator or end user.
(If the operator does not understand it the feature gate/admission plugin will not be enabled. In case the end user does not understand it the `commonSuffixes` will not be set correctly or left completely blank.)
Splitting this documentation would result in a more or less common part and only small parts with changes. If we split it in three documents (common/operator/end user) one document (common) will be lengthy while the others would be fairly small.
Therefore, I would leave it as is unless you have a strong opinion otherwise.